src/index.js

```diff
@@ -110,81 +107,99 @@ function SpanCreator({ tracer, serviceName, kind }) {
     }
 
     constructor(spanName, options) {
-      const id = this._getTraceId(options);
-      this.id = id;
-
-      if (!this._constructedFromOutside(options)) {
-        tracer.scoped(() => {
-          tracer.setId(id);
-          if (spanName) {
-            tracer.recordAnnotation(new Annotation.Rpc(spanName));
-          }
+      super();
 
-          tracer.recordServiceName(serviceName);
-          tracer.recordAnnotation(new startSpanAnnotation[kind]());
-        });
-      }
+      this.id = this._getTraceId(options);
+
+      tracer.scoped(() => {
+        tracer.setId(this.id);
+        if (spanName) {
+          tracer.recordAnnotation(new Annotation.Rpc(spanName));
+        }
+
+        tracer.recordServiceName(serviceName);
+        tracer.recordAnnotation(new startSpanAnnotation[kind]());
+      });
+    }
+
+    _tracer() {
+      return otTracer;
     }
+    _context() {
+      return this.id;
+    }
+
+    // TODO: ??? _setOperationName(newName) { this._name = newName; }
```

Comment:
This should be as simple as setting a default value for `startSpan`, right?

Reply:
I don't know, if you can change the `span` name in `zipkin` after it has been set for the first time.

Reply:
No, I don't think so